test/js_test.go

```diff
@@ -593,17 +593,22 @@ func TestJetStreamSubscribe(t *testing.T) {
 
 	// If we are here we have received all of the messages.
 	// We hang the ConsumerInfo option off of the subscription, so we use that to check status.
-	info, _ := sub3.ConsumerInfo()
-	if info.Config.AckPolicy != nats.AckExplicitPolicy {
-		t.Fatalf("Expected ack explicit policy, got %q", info.Config.AckPolicy)
-	}
-	if info.Delivered.Consumer != uint64(toSend) {
-		t.Fatalf("Expected to have received all %d messages, got %d", toSend, info.Delivered.Consumer)
-	}
-	// Make sure we auto-ack'd
-	if info.AckFloor.Consumer != uint64(toSend) {
-		t.Fatalf("Expected to have ack'd all %d messages, got ack floor of %d", toSend, info.AckFloor.Consumer)
-	}
+	// We may need to retry this check since the acks sent by the client have to be processed
+	// on the server.
+	checkFor(t, 2*time.Second, 100*time.Millisecond, func() error {
```

Comment:
For sake of slow CI runs, maybe it's better to increase the max Duration to 10 seconds?